src/rust_connection/mod.rs

```diff
@@ -138,7 +129,7 @@ impl RustConnection<BufReadFD<stream::Stream>, BufWriteFD<stream::Stream>> {
     }
 }
 
-impl<R: ReadFD, W: WriteFD> RustConnection<R, W> {
+impl<R: ReadFD + Poll, W: WriteFD + Poll> RustConnection<R, W> {
```

Comment:
Some doc comment somewhere should clarify the behaviour of `R` when polled for writability and `W` for readability. The code assumes that the two know about each other and "poll for each other".

Reply:
Yes, this needs to be documented. There is also some inconsistent behavior for buffered streams, because the reader cannot access the writer buffer stream and vice versa.

I'm thinking on how to implement this with a single object that allows reading + writing.